worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala

```diff
@@ -354,9 +354,17 @@ private[deploy] class Controller(
 
     val shuffleCommitTimeout = conf.workerShuffleCommitTimeout
 
-    shuffleCommitInfos.putIfAbsent(shuffleKey, JavaUtils.newConcurrentHashMap[Long, CommitInfo]())
+    shuffleCommitInfos.computeIfAbsent(
+      shuffleKey,
+      (k: String) => {
+        return JavaUtils.newConcurrentHashMap[Long, CommitInfo]()
+      })
     val epochCommitMap = shuffleCommitInfos.get(shuffleKey)
-    epochCommitMap.putIfAbsent(epoch, new CommitInfo(null, CommitInfo.COMMIT_NOTSTARTED))
+    epochCommitMap.computeIfAbsent(
+      epoch,
+      (k: String) => {
+        return new CommitInfo(null, CommitInfo.COMMIT_NOTSTARTED)
+      })
```

Comment:
ditto, in several places
```suggestion
      (k: String) => new CommitInfo(null, CommitInfo.COMMIT_NOTSTARTED))
```